File: 0739-daily-temperatures/0739-daily-temperatures.js
/**
 * @param {number[]} temperatures
 * @return {number[]}
 */
var dailyTemperatures = function(temperatures) {
    var lengthh = temperatures.length;
    var answer = [];
    for(j=0; j < lengthh; j++){
        answer.push(0);
    }
    
    var monStack = [];
    
    for(i=0; i < lengthh; i++){
        while(monStack && temperatures[monStack[monStack.length-1] ]< temperatures[i]){
            let prev = monStack.pop();
            answer[prev] = (i - prev);
        }
        monStack.push(i);
    }
    return answer
};

File: 0739-daily-temperatures/0739-daily-temperatures.py
class Solution:
    def dailyTemperatures(self, temperatures: List[int]) -> List[int]:
        
        ans = [0]*len(temperatures)
        mon_stack = []
        
        for i in range(len(temperatures)):
            while mon_stack and temperatures[mon_stack[-1]] < temperatures[i]:
                curr = mon_stack.pop()
                ans[curr] = (i-curr)
                
            mon_stack.append(i)
            
        return ans